libs/langchain/langchain/llms/triton_tensorrt.py

```diff
@@ -0,0 +1,537 @@
+        return inputs
+
+    def _generate_outputs(
+        self,
+    ) -> List[Union[grpcclient.InferRequestedOutput, httpclient.InferRequestedOutput]]:
+        """Generate the expected output structure."""
+        return [self._infer_output("text_output")]
+
+    def _prepare_tensor(
+        self, name: str, input_data: Any
+    ) -> Union[grpcclient.InferInput, httpclient.InferInput]:
+        """Prepare an input data structure."""
+        from tritonclient.utils import np_to_triton_dtype
+
+        t = self._infer_input(
+            name, input_data.shape, np_to_triton_dtype(input_data.dtype)
+        )
+        t.set_data_from_numpy(input_data)
+        return t
+
+    def _generate_inputs(  # pylint: disable=too-many-arguments,too-many-locals
+        self,
+        prompt: str,
+        tokens: int = 300,
+        temperature: float = 1.0,
+        top_k: float = 1,
+        top_p: float = 0,
+        beam_width: int = 1,
+        repetition_penalty: float = 1,
+        length_penalty: float = 1.0,
+        stream: bool = True,
+    ) -> List[Union[grpcclient.InferInput, httpclient.InferInput]]:
+        """Create the input for the triton inference server."""
+        query = np.array(prompt).astype(object)
+        request_output_len = np.array([tokens]).astype(np.uint32).reshape((1, -1))
+        runtime_top_k = np.array([top_k]).astype(np.uint32).reshape((1, -1))
+        runtime_top_p = np.array([top_p]).astype(np.float32).reshape((1, -1))
+        temperature_array = np.array([temperature]).astype(np.float32).reshape((1, -1))
+        len_penalty = np.array([length_penalty]).astype(np.float32).reshape((1, -1))
+        repetition_penalty_array = (
+            np.array([repetition_penalty]).astype(np.float32).reshape((1, -1))
+        )
+        random_seed = np.array([RANDOM_SEED]).astype(np.uint64).reshape((1, -1))
+        beam_width_array = np.array([beam_width]).astype(np.uint32).reshape((1, -1))
+        streaming_data = np.array([[stream]], dtype=bool)
+
+        inputs = [
+            self._prepare_tensor("text_input", query),
+            self._prepare_tensor("max_tokens", request_output_len),
+            self._prepare_tensor("top_k", runtime_top_k),
+            self._prepare_tensor("top_p", runtime_top_p),
+            self._prepare_tensor("temperature", temperature_array),
+            self._prepare_tensor("length_penalty", len_penalty),
+            self._prepare_tensor("repetition_penalty", repetition_penalty_array),
+            self._prepare_tensor("random_seed", random_seed),
+            self._prepare_tensor("beam_width", beam_width_array),
+            self._prepare_tensor("stream", streaming_data),
+        ]
+        return inputs
+
+    def _trim_batch_response(self, result_str: str) -> str:
+        """Trim batch response by removing prompt and extra generated text."""
+        # extract the generated part of the prompt
+        split = result_str.split("[/INST]", 1)
```

Comment:
This seems like it assumes Llama-style prompt formatting.. would be great to make this less rigid